auto_round/__main__.py

```diff
@@ -0,0 +1,472 @@
+    )
+
+    parser.add_argument('--eval', action='store_true', 
+                        help="whether to use eval mode.")
+
+    parser.add_argument("--bits", default=4, type=int,
+                        help="number of  bits")
+
+    parser.add_argument("--group_size", default=128, type=int,
+                        help="group size")
+
+    parser.add_argument("--batch_size", default=8, type=int,
+                        help="train batch size")
+
+    parser.add_argument("--eval_bs", default=1, type=int,
+                        help="eval batch size")
+
+    parser.add_argument("--device", default="auto", type=str,
+                        help="The device to be used for tuning. The default is set to auto/None,"
+                             "allowing for automatic detection. Currently, device settings support CPU, GPU, and HPU.")
+
+    parser.add_argument("--sym", action='store_true',
+                        help=" sym quantization")
+
+    parser.add_argument("--iters", default=200, type=int,
+                        help=" iters")
+
+    parser.add_argument("--dataset", default="NeelNanda/pile-10k", type=str,
+                        help="The dataset for quantization training. It can be a custom one.")
+
+    parser.add_argument("--enable_quanted_input", action='store_true',
+                        help="enable_quanted_input is deprecated.")
+
+    parser.add_argument("--lr", default=None, type=float,
+                        help="learning rate, if None, it will be set to 1.0/iters automatically")
+
+    parser.add_argument("--minmax_lr", default=None, type=float,
+                        help="minmax learning rate, if None,it will beset to be the same with lr")
+
+    parser.add_argument("--seed", default=42, type=int,
+                        help="seed")
+
+    parser.add_argument("--amp", action='store_true',
+                        help="amp is deprecated ")
+
+    parser.add_argument("--adam", action='store_true',
+                        help="adam")
+
+    parser.add_argument("--seqlen", default=2048, type=int,
+                        help="sequence length")
+
+    parser.add_argument("--gradient_accumulate_steps", default=1, type=int, help="gradient accumulate steps")
+
+    parser.add_argument("--nblocks", default=1, type=int, help="num of blocks to tune together")
+
+    parser.add_argument("--nsamples", default=128, type=int,
+                        help="number of samples")
+
+    parser.add_argument("--low_gpu_mem_usage", action='store_true',
+                        help="lower gpu memory but 50%-100% slower")
+
+    parser.add_argument("--enable_minmax_tuning", action='store_true',
+                        help="enable_minmax_tuning is deprecated")
+
+    parser.add_argument("--format", default=None, type=str,
+                        help="targeted inference acceleration platform,The options are 'fake', 'cpu', 'gpu' and 'xpu'."
+                            "default to 'fake', indicating that it only performs"
```

Comment:
format should be auto_round, auto_awq, auto_gptq, etc.. Adjust the save code accordingly to reflect these changes. If you need assistance, feel free to reach out to Weiwei for help.

In the example code, we maintain both deploydevice and format for compatibility. However, inform users that deploydevice is deprecated. It will still function, since some recipes are using it.